src/StatsdClient/StatsBufferize.cs

```diff
@@ -0,0 +1,76 @@
+using System.Threading.Tasks;
+using System;
+using StatsdClient.Worker;
+
+namespace StatsdClient
+{
+    /// <summary>
+    /// StatsBufferize bufferizes metrics before sending them.
+    /// </summary>
+    class StatsBufferize : IStatsdUDP, IDisposable
+    {
+        readonly AsynchronousWorker<string> _worker;
+
+        public StatsBufferize(
+            BufferBuilder bufferBuilder,
+            int workerMaxItemCount,
+            TimeSpan? blockingQueueTimeout,
+            TimeSpan maxIdleWaitBeforeSending)
+        {
+            var handler = new WorkerHandler(bufferBuilder, maxIdleWaitBeforeSending);
+
+            // `handler` (and also `bufferBuilder`) do not need to be thread safe as long as workerMaxItemCount is 1.
+            this._worker = new AsynchronousWorker<string>(handler, 1, workerMaxItemCount, blockingQueueTimeout);
+        }
+
+        public void Send(string command)
+        {
+            this._worker.TryEnqueue(command);
+        }
+
+        public void Dispose()
+        {
+            this._worker.Dispose();
+        }
+
+        public Task SendAsync(string command)
+        {
+            throw new NotSupportedException();
```

Comment:
Note: `SendAsync` will be marked as deprecated in the PR.

Reply:
We have to think twice about breaking changes..... we should maybe discuss with the rest of the team. If we can help users avoid making changes to their code, that'd be ideal.

Reply:
We decided with the team to mark some methods as deprecated instead of making breaking changes.

`StatsBufferize` is not public and so cannot be instantiated by the user.  The method `SendAsync` is not reachable but as `StatsBufferize` implements `IStatsdUDP` I have to provide an implementation. 

Note: `SendAsync` will be marked as deprecated in a future PR.